src/coreclr/jit/gentree.cpp

```diff
@@ -5977,7 +5977,9 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
             if (call->gtCallType == CT_INDIRECT)
             {
                 // pinvoke-calli cookie is a constant, or constant indirection
-                assert(call->gtCallCookie == nullptr || call->gtCallCookie->OperIs(GT_CNS_INT, GT_IND));
+                // or a non-tree if this is a managed call.
+                assert(call->gtCallCookie == nullptr || call->gtCallCookie->OperIs(GT_CNS_INT, GT_IND) ||
+                       call->IsVirtualStub());
```

Comment:
This one seems especially problematic, aren't we going to be treating a `HandleHistogramProfileCandidateInfo*` or `InlineCandidateInfo*` as a `GenTree*` and calling `OperIs` on it?